packages/nitro-protocol/docs/force-move-safety.md

```diff
@@ -0,0 +1,399 @@
+
+## `respondAlt`
+
+This action allows participants to provide the adjudicator an alternative set of states that also increases the `turnNumRecord`.
+Participants may wish to sign multiple states with the same `turnNumRecord` for various practical purposes.
+One such purpose is to allow Alice to send money to Bob while Bob is off-line, by sending multiple states at turn `n` with increasing amounts of money allocated to Bob.
+Bob may then reply to whatever state is most desirable to him.
+
+To apply `respondAlt(n)`, the participant must supply a supported state `s` with turn number `n`.
+In other words, they supply a pair `(s, s')` of states, where
+
+- `s'.turnNumber == n`
+- `validTransition(s, s') == true`
+- `s` and `s'` are both signed by the correct participant
+
+## `refute(n, s)`
+
+To apply this action, a participant can simply supply a correctly-signed state `s`.
+When applying `refute(n, s)`, the turn number of the provided state `s` must be `n`.
+
+This action is meant to prove that a participant maliciously called `forceMove` with a knowingly stale state.
+The goal is to provide a state, signed by the challenger, with a greater turn number than the challenge state.
+If the challenger signed such a state, the adjudicator can infer that the challenger should be aware of a later supported state, and discards the current challenge.
+At this point, the adjudicator could penalize the challenger for malicious intent.
+
+## `timeout`
+
+Whenever the state of the channel is updated to `S`, the adjudicator makes a note of `S`, and starts a timer.
+When the timer runs out, the adjudicator applies `timeout(S)`.
+
+In practice, this transition happens implicitly: when in a `Challenge` state, the challenge expiration time is recorded in a field `finalizesAt`, and once `finalizesAt <= now`, the channel is closed.
+
+## State Transitions, V1
+
+We record the allowed state transitions from the original ForceMove protocol in the following table.
+By default, any (state, action) pair not listed below does not change the state.
+
+| State       | Action           | Next state  | Requiremeents   |
+| ----------- | ---------------- | ----------- | --------------- |
+| Open(n)     | forceMove(m,s,p) | Chal(m,s,p) | m >= n          |
+| Chal(n,s,p) | respond(s')      | Open(n+1)   | s -> s'         |
+| Chal(n,s,p) | respondAlt(n+1)  | Open(n+1)   |                 |
+| Chal(n,s,p) | refute(m,s')     | Open(n)     | m > n           |
+|             |                  |             | signer(p') == p |
+| Chal(n,s,p) | timeout(S)       | Closed(s)   | matches(S)      |
+
+Here, `matches(S)` indicates that the current on-chain channel state is `S`.
+`Chal` is used as a short-hand for `Challenge`.
+In other words, applying `timeout` to a `Challenge` state only has an effect if the state has not changed since the adjudicator timer has started.
+Applying `timeout` to an open state has no effect, as the transition is not listed in this table.
```

Comment:
It would be great to disambiguate the state of the _channel_ from the state of the _adjudicator_ throughout. For example, "open adjudicator state" and so on.

Reply:
I've done this partway. I'm 50/50 on if "state" needs any further clarification in the phrase "`Open` state". I'm almost thinking about introducing some variable called `AdjudicatorState` to be completely explicit, since there are so many different types of states discussed in the post.

Reply:
Yeah it's a judgement call on whether the context is enough to prevent confusion, or whether more terminology helps. I tend to prefer the more explicit route, because it is easy to underestimate  the confusion for those who don't look at this every day.